src/compiler/commandLineParser.ts

```diff
@@ -3083,19 +3152,35 @@ namespace ts {
                 currentOption[option.name] = normalizeOptionValue(option, basePath, value);
             },
             onSetValidOptionKeyValueInRoot(key: string, _keyNode: PropertyName, value: CompilerOptionsValue, valueNode: Expression) {
-                switch (key) {
-                    case "extends":
-                        const newBase = configFileName ? directoryOfCombinedPath(configFileName, basePath) : basePath;
-                        extendedConfigPath = getExtendsConfigPath(
-                            value as string,
-                            host,
-                            newBase,
-                            errors,
-                            (message, arg0) =>
-                                createDiagnosticForNodeInSourceFile(sourceFile, valueNode, message, arg0)
-                        );
-                        return;
-                }
+                    switch (key) {
+                        case "extends":
+                            const newBase = configFileName ? directoryOfCombinedPath(configFileName, basePath) : basePath;
+                            if (isString(value)) {
+                                extendedConfigPath = append (extendedConfigPath, getExtendsConfigPath(

```

Comment:
Making this change would break the `adds extendedSourceFiles from an array only once:' and `can overwrite top-level compilerOptions with jsonSourceFile` because the extended path comes to be only the path of the file that is extending the other files and does not include the compiler options or the path of the files being extended.